### PR TITLE
[feat] Add network with NodePort step

### PR DESCRIPTION
**Pull request by AlexandreBrg**
## Context

Implement a start for the network in the scheduler 

## What is in this PR

* When a function is created, a random port is defined `45000..50000`
Closes https://github.com/polyxia-org/polyxia-org/issues/22

### CODE REVIEW

File: crates/definition/src/lib.rs [hunk 1]
```diff
@@ -47,6 +47,18 @@ pub mod workload {
         pub port_type: NetworkPortExposureType,
     }
 
+    impl FunctionPort {
+        /// Create a FunctionPort and bind it to the default port 3000
+        /// All our runtimes only use this port
+        pub fn new(port: u16) -> Self {
+            Self {
+                port,
+                target_port: 3000,
```
Comment (hugoamalric) [inline crates/definition/src/lib.rs:56]: Rather put it in a variable if it ever changes

**Review by hugoamalric — CHANGES_REQUESTED** [review]

**Comment by mflagey** [comment]
Can you fix warnings please ?

File: crates/definition/src/lib.rs (hunk 1 above)
Comment (AlexandreBrg) [reply crates/definition/src/lib.rs]: Done

**Comment by AlexandreBrg** [comment]
> Can you fix warnings please ?

@mflagey Applied

**Review by mflagey — APPROVED** [review]

**Pull request merged**